File: chrome/test/data/webui/chromeos/shortcut_customization/shortcut_customization_browsertest.js
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview Test suite for chrome://shortcut-customization. Tests
 * individual polymer components in isolation. To run all tests in a single
 * instance (default, faster):
 * `browser_tests --gtest_filter=ShortcutCustomizationApp*`
 * To run a single test suite such as 'AcceleratorRowTest':
 * browser_tests --gtest_filter=ShortcutCustomizationAppAcceleratorRowTest.All
 */

GEN_INCLUDE(['//chrome/test/data/webui/polymer_browser_test_base.js']);

GEN('#include "ui/base/ui_base_features.h"');
GEN('#include "content/public/test/browser_test.h"');
GEN('#include "ash/constants/ash_features.h"');
GEN('#include "chromeos/constants/chromeos_features.h"');

var ShortcutCustomizationAppBrowserTest = class extends PolymerTest {
  get browsePreload() {
    return 'chrome://shortcut-customization/';
  }

  get featureList() {
    return {
      enabled: [
        'features::kShortcutCustomizationApp',
        'features::kShortcutCustomization',
        // TODO(b/276493795): Remove jelly and
        // shortcut-customization-jelly after the Jelly experiment is launched.
        'chromeos::features::kJelly',
        'ash::features::kShortcutCustomizationJelly',
      ]
    };
  }
};

const tests = [
  ['AcceleratorEditViewTest', 'accelerator_edit_view_test.js'],
  ['AcceleratorLookupManagerTest', 'accelerator_lookup_manager_test.js'],
  ['AcceleratorViewTest', 'accelerator_view_test.js'],
  ['AcceleratorRowTest', 'accelerator_row_test.js'],
  ['AcceleratorEditDialogTest', 'accelerator_edit_dialog_test.js'],
  ['AcceleratorSubsectionTest', 'accelerator_subsection_test.js'],
  ['BottomNavContentTest', 'bottom_nav_content_test.js'],
  ['FakeShortcutProviderTest', 'fake_shortcut_provider_test.js'],
  ['FakeShortcutSearchHandlerTest', 'fake_shortcut_search_handler_test.js'],
  ['InputKeyTest', 'input_key_test.js'],
  ['RouterTest', 'router_test.js'],
  ['SearchBoxTest', 'search_box_test.js'],
  ['SearchResultRowTest', 'search_result_row_test.js'],
  ['SearchResultBoldingTest', 'search_result_bolding_test.js'],
  ['ShortcutCustomizationApp', 'shortcut_customization_test.js'],
  ['ShortcutSearchHandlerTest', 'shortcut_search_handler_test.js'],
  ['ShortcutsPageTest', 'shortcuts_page_test.js'],
  ['ShortcutUtils', 'shortcut_utils_test.js'],
  ['TextAcceleratorTest', 'text_accelerator_test.js'],
];

tests.forEach(test => registerTest(...test));

function registerTest(testName, module, caseName) {
  const className = `ShortcutCustomizationApp${testName}`;
  this[className] = class extends ShortcutCustomizationAppBrowserTest {
    /** @override */
    get browsePreload() {
      return `chrome://shortcut-customization/test_loader.html` +
          `?module=chromeos/shortcut_customization/${module}`;
    }
  };

  TEST_F(className, caseName || 'All', () => mocha.run());
}

File: chrome/test/data/webui/chromeos/shortcut_customization/shortcut_customization_browsertest.cc
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <string>

#include "ash/webui/shortcut_customization_ui/url_constants.h"
#include "base/strings/stringprintf.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/test/browser_test.h"

/**
 * @fileoverview Test suite for chrome://shortcut-customization.
 */

namespace ash {

namespace {

class ShortcutCustomizationAppBrowserTest : public WebUIMochaBrowserTest {
 public:
  ShortcutCustomizationAppBrowserTest() {
    set_test_loader_host(::ash::kChromeUIShortcutCustomizationAppHost);
  }

 protected:
  void RunTestAtPath(const std::string& testFilePath) {
    auto testPath = base::StringPrintf("chromeos/shortcut_customization/%s",
                                       testFilePath.c_str());
    WebUIMochaBrowserTest::RunTest(testPath, "mocha.run()");
  }
};

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       AcceleratorEditViewTest) {
  RunTestAtPath("accelerator_edit_view_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       AcceleratorLookupManagerTest) {
  RunTestAtPath("accelerator_lookup_manager_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       AcceleratorRowTest) {
  RunTestAtPath("accelerator_row_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       AcceleratorEditDialogTest) {
  RunTestAtPath("accelerator_edit_dialog_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       AcceleratorSubsectionTest) {
  RunTestAtPath("accelerator_subsection_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       BottomNavContentTest) {
  RunTestAtPath("bottom_nav_content_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       FakeShortcutProviderTest) {
  RunTestAtPath("fake_shortcut_provider_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       FakeShortcutSearchHandlerTest) {
  RunTestAtPath("fake_shortcut_search_handler_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest, RouterTest) {
  RunTestAtPath("router_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest, SearchBoxTest) {
  RunTestAtPath("search_box_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       SearchResultRowTest) {
  RunTestAtPath("search_result_row_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       SearchResultBoldingTest) {
  RunTestAtPath("search_result_bolding_test.js");
}

// TODO(crbug.com/369506934): De-flake and re-enable
IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       DISABLED_ShortcutCustomizationApp) {
  RunTestAtPath("shortcut_customization_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       ShortcutSearchHandlerTest) {
  RunTestAtPath("shortcut_search_handler_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest, ShortcutsPageTest) {
  RunTestAtPath("shortcuts_page_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest, ShortcutUtils) {
  RunTestAtPath("shortcut_utils_test.js");
}

IN_PROC_BROWSER_TEST_F(ShortcutCustomizationAppBrowserTest,
                       TextAcceleratorTest) {
  RunTestAtPath("text_accelerator_test.js");
}

}  // namespace

}  // namespace ash
